### ISSUE TITLE
[GeoMechanicsApplication] Make sure the structural/interface elements used in the quay wall work with the geo reset displacement process

### ISSUE THREAD

**Issue by rfaasse** [KratosMultiphysics/Kratos#12868]
This holds for the following elements:
- LinearTimoshenkoCurvedBeamElement2D3N
- LinearTimoshenkoBeamElement2D3N
- LinearTimoshenkoBeamElement2D2N
- TrussElement3D2N
- TrussElementLinear3D2N
- LinearTrussElement2D (two and three nodes, but this shouldn't add extra work)
- Our new interface element

This means we need to make sure:
- All elements can calculate the PK2_STRESS_VECTOR (does this make sense, or should the reset displacement process be able to handle different stresses?)
- Make sure their respective constitutive laws use the InitialState to calculate their material response
- Make sure these elements work with the 'rest' input, especially pay attention to 'rest' in combination with excavation

**Comment by indigocoral** [comment]
@markelov208 Could you please specify the error you face while running this?

**Comment by markelov208** [comment]
@indigocoral Anne told that Richard fixed this in his branch. The problem is the following. Kratos reads the entire model. Some elements are used at the first stage but some of them will be enabled only at the second stage. However, Check function can be run for these unenabled elements.  They were not initialized; therefore, mConstitutiveLawVector has a zero size which causes a crash down.

**Comment by indigocoral** [comment]
@markelov208 Thank you very much.
FYI for clarity on the status of this issue @avdg81 @rfaasse @mcgicjn2

**Comment by rfaasse** [comment]
Indeed, the issue that @markelov208 described, I have fixed in a branch, I'll create a PR for it, since it's a sensible thing to add for the structural elements (meaning I expect it will be accepted quite quickly). 

There is one more issue related to excavation + initialization + the "rest" input mode, but we want to have a look at the orchestrator in #13044 to see if that could fix our issue instead of making a PR for structural with a hacky implementation. 

Therefore, I would propose to close this issue after the PR that I mentioned at the start is merged and create a follow-up (or use #13044 as part of the follow-up) for the remaining issue. Otherwise, this one could stay open for a long time
